lightning/src/util/test_utils.rs

```diff
@@ -201,6 +202,7 @@ impl<Signer: keysinterface::Sign> channelmonitor::Persist<Signer> for TestPersis
 
 pub struct TestBroadcaster {
 	pub txn_broadcasted: Mutex<Vec<Transaction>>,
+	pub blocks: Arc<Mutex<Vec<(BlockHeader, u32)>>>,
```

Comment:
Likewise, why is an `Arc` needed?

Reply:
Hmm, not sure what the "Likewise" here is, but `TestBroadcaster` gets passed into various structs with a reference to the same blocks as the `Node` object holds. It failed to compile without being thread-safe (ie arc+mutex), but I didn't investigate further as it seemed sensible that it wouldn't work with an Rc.